### ISSUE TITLE
Having an @ in your message acts weird while sending messages

### ISSUE THREAD

**Issue by PureTryOut** [cuberite/cuberite#5287]
Client version: 1.12.2
Server OS: Linux
Cuberite Commit id: 4b8952e438da8e57d6442a0b7451dac8060f154b

### Expected behavior
When not using commands, `@` and any characters after it should appear as normal in messages.

### Actual behavior
Having an `@` in the message causes weird behaviour. E.g. sending the message `@t` causes an empty message to be send, but sending just the `@` does send that character. Sending something with `@d` makes the rest of the text appear pink, so I'm guessing these are color codes somehow? It's quite unexpected at least.

As for use-case, I was sharing a [Matrix](https://matrix.org) address with someone and those are in the format of `@username:server.domain`. Sharing e-mail addresses is also quite hard. Of course most people won't do this, but it should still be possible.

The full message (with `@`) character is shown correctly in the console however.

### Steps to reproduce the behavior
1. Send any message with an @ and some character after it immediately.

**Comment by PureTryOut** [comment]
I uh, don't think it's really fixed. Nice that it doesn't do weird things for people without the `chat.format` permissions anymore but people that _do_ have that permission still have the strange behaviour. I was still confused for a second why it was happening on my personal server at least. Could the formatting be changed to use Bukket-like color codes with `&`?

**Comment by LittleBigBug** [comment]
Yes, This should be closed with #4501, I think

**Comment by PureTryOut** [comment]
Yeah seems like it is, till that is merged this issue should be re-opened.

**Comment by tonitch** [comment]
> `&` does seem like a better character already, especially if alternative server implementations use that.

isn't there a problem for http link tho? I think `$` seems like a betterer replacement ?

Either we check if the message is a link and don't parse it or we use another character I think!

**Comment by tonitch** [comment]
it seems like in the current version the `§` is the replacement of the previous version, the `&` character. (for config file pretty much. this should not work in the chat)
however, most server implement essentials_chat or an equivalent that will make the `&a` style character work as this has become a "standard"

I search a bit on the essentialsx plugin wiki and can't find any way that they escape the & character... I think we can do in the standard again and go for a `\&` when we don't want the color code to apply

Also, maybe this could be a plugin as some server maybe don't want the color code to apply on any message at all?

So the cCompositeChat would have to receive real color code (`§` I guess because it doesn't appear on much keyboard anyway) and a plugin would just replace any `&x` valid color code with any `§x` valid color code

does this seems like a good implementation to you ?

source : https://minecraft.fandom.com/wiki/Formatting_codes
